src/callbacks_stage/subcell_bounds_check.jl

```diff
@@ -0,0 +1,124 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+"""
+    BoundsCheckCallback(; output_directory="out", save_errors=false, interval=1)
+
+Bounds checking routine for [`SubcellLimiterIDP`](@ref). Applied as a stage callback for SSPRK
+methods. If `save_errors` is `true`, the resulting deviations are saved in
+`output_directory/deviations.txt` for every `interval` time steps.
+
+!!! note
+    For `SubcellLimiterIDP`, the solution is corrected in the a posteriori correction stage
+    [`SubcellLimiterIDPCorrection`](@ref). So, to check the final solution, this bounds check
+    callback must be called after the correction stage.
+"""
+struct BoundsCheckCallback
+    output_directory::String
+    save_errors::Bool
+    interval::Int
+end
+
+function BoundsCheckCallback(; output_directory = "out", save_errors = false,
+                             interval = 1)
+    BoundsCheckCallback(output_directory, save_errors, interval)
+end
+
+function (callback::BoundsCheckCallback)(u_ode, integrator, stage)
+    mesh, equations, solver, cache = mesh_equations_solver_cache(integrator.p)
+    (; t, iter, alg) = integrator
+    u = wrap_array(u_ode, mesh, equations, solver, cache)
+
+    save_errors = callback.save_errors && (callback.interval > 0) &&
+                  (stage == length(alg.c)) &&
```

Comment:
Right now, if `save_errors=true` only the errors of the last stage of the specific RK method are saved. @amrueda Do you think it would help and is worth spending time on the implementation to generalize this.

I could add a parameter with all stage number which should be saved. Actually, I think that should be very easy to implement.
And I would just save the last stage as a default setting.

Reply:
It actually would have been helpful for our debugging sessions, but I don't know if that's still the case.

Reply:
I don't think that it is necessary to *save* the errors at every RK stage. However, it would be useful to know if the bounds were fulfilled always.
Would it be easy to record the maximum deviation from the bounds at all RK stages and then save that maximum deviation at the end of each time step? (or every `n` time steps)... That way the user knows if the bounds were always met.

Reply:
This is exactly how it is currently implemented:muscle: Thank you.

Reply:
After discussing this with @amrueda, we decided that it would be more useful to export the maximum deviations since the last export instead of just saving the error from the last stage.
Right now, errors from e.g. RK stage 1 and 2 are not visible in the `deviations.txt` file. Additionally, if `interval>1` more deviations are invisible.

Reply:
Done with this [PR](https://github.com/bennibolm/Trixi.jl/pull/117).